src/StackAdmin/Azs.Fabric.Admin/Tests/src/StorageSubSystem.tests.ps1

```diff
@@ -0,0 +1,163 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+    Run AzureStack fabric admin storage subsystem tests.
+
+.DESCRIPTION
+    Run AzureStack fabric admin storage subsystem tests using either mock client or our client.
+	The mock client allows for recording and playback.  This allows for offline tests.
+
+.PARAMETER RunRaw
+    Run using our client creation path.
+
+.EXAMPLE
+    PS C:\> .\src\StorageSubSystem.Tests.ps1
+    Describing StorageSubSystems
+     [+] TestListStorageSubSystems 133ms
+     [+] TestGetStorageSubSystem 63ms
+     [+] TestGetAllStorageSubSystems 84ms
+
+.EXAMPLE
+    PS C:\> .\src\StorageSubSystem.Tests.ps1 -RunRaw $true
+    Describing StorageSubSystems
+     [+] TestListStorageSubSystems 1.83s
+     [+] TestGetStorageSubSystem 1.96s
+     [+] TestGetAllStorageSubSystems 2.09s
+
+.NOTES
+    Author: Yuxing Zhou
+	Copyright: Microsoft
+    Date:   October 31, 2018
+#>
+param(
+    [bool]$RunRaw = $false,
+    [bool]$UseInstalled = $false
+)
+
+$Global:UseInstalled = $UseInstalled
+$global:RunRaw = $RunRaw
+$global:TestName = ""
+
+. $PSScriptRoot\CommonModules.ps1
+
+InModuleScope Azs.Fabric.Admin {
+
+    Describe "StorageSubSystems" -Tags @('StorageSubSystem', 'Azs.Fabric.Admin') {
+
+        . $PSScriptRoot\Common.ps1
+
+        BeforeEach {
+
+            function ValidateStorageSubSystem {
+                param(
+                    [Parameter(Mandatory = $true)]
+                    $StorageSubSystem
+                )
+
+                $StorageSubSystem          | Should Not Be $null
+
+                # Resource
+                $StorageSubSystem.Id       | Should Not Be $null
+                $StorageSubSystem.Location | Should Not Be $null
+                $StorageSubSystem.Name     | Should Not Be $null
+                $StorageSubSystem.Type     | Should Not Be $null
+
+                # Storage Subsystem
+                $StorageSubSystem.TotalCapacityGB      | Should Not Be $null
+                $StorageSubSystem.RemainingCapacityGB  | Should Not Be $null
+                $StorageSubSystem.HealthStatus         | Should Not Be $null
```

Comment:
Is this an ENUM?  If so you should probably test the value to ensure that it is valid.

Reply:
Also for OperationalStatus and RebalanceStatus.

Reply:
No, they are string.